api/src/clojure/org/akvo/flow_api/unilog/unilog.clj

```diff
@@ -115,22 +115,24 @@
                                                survey-changed
                                                survey-deleted]}
                                       form-id->form
-                                      authorized-forms
-                                      authorized-surveys]
+                                      survey-id->survey]
   {:unilog-id unilog-id
    :form-changed (->> form-updated (keep form-id->form) set)
-   :form-deleted (set/intersection authorized-forms form-deleted)
+   :form-deleted form-deleted
    :form-instance-deleted form-instance-deleted
    :form-instances-to-load (->> forms-instances-grouped-by-form
                                 (keep (fn [[form-id form-instance]]
                                         (when-let [form (get form-id->form form-id)]
                                           {:form form
                                            :form-instance-ids (set (map :id form-instance))})))
-                                set) ;; the mapping form-id->form filter instances
-   :data-point-changed (filter #(authorized-surveys (:survey-id %)) data-point-changed)
+                                set)
+   :data-point-changed (->> data-point-changed
+                            (filter (comp (set (keys survey-id->survey)) :survey-id))
```

Comment:
`(set (keys survey-id->survey))` == `survey-id->survey` for this particular case.

A map is a fn.